### PR TITLE
Create requirements file and update python dependencies

### LINKED ISSUE #1072: Add requirements.txt and review wazuh_deps setup
We would require to have a `requirements.txt` file with all the pip dependencies required for running our tests. This file would be updated every time someone adds a new dep.

Also, we would have to check that wazuh_deps's `setup.py` installs all the package dependencies and make sure everyone updates this script when adds a new dependency to the package code.

### LINKED ISSUE #1095: Update Python dependencies for integration tests
Gcloud integration tests are failing on Jenkins due to missing dependencies, we need to check it out and fix it.

Example of a failed build. https://ci.wazuh.info/job/test_integration/

image

### PR DESCRIPTION
**Pull request by Rebits**
|Related issue|
|---|
|Closes #1072 , Closes #1095|

## Description

A requirements file have been created, replacing `install_requires` in setup script. Also, setting up test environment documentation has been updated.

## Documentation

setting_up

### CODE REVIEW

**Review by jmv74211 — CHANGES_REQUESTED** [review]
You have to open an issue in wazuh CICD to install new requirements txt file during the provisioning.

File: docs/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
 FROM python
 
-COPY ./requirements.txt /
+COPY ../../requirements.txt /
```
Comment (BraulioV) [inline docs/Dockerfile:3]:
```suggestion
COPY ../requirements.txt /
```

File: docs/tests/integration/setting_up_test_environment.md [hunk 2]
```diff
@@ -20,30 +25,27 @@ _We are using **CentOS** for this example:_
...
 - Disable firewall (only for **CentOS**)
 
-```shell script
+```
 systemctl stop firewalld
 systemctl disable firewalld
 ```
 
-- Install Python and its dependencies
+- Install Python and development tools
```
Comment (BraulioV) [inline docs/tests/integration/setting_up_test_environment.md:33]:
```suggestion
- Install Python and the development tools
```

File: docs/tests/integration/setting_up_test_environment.md [hunk 3]
```diff
@@ -60,12 +62,15 @@ echo 'monitord.rotate_log=0' >> $wazuh_path/etc/local_internal_options.conf
 /var/ossec/bin/ossec-control restart
 ```
 
-- Download wazuh-qa repository and install package dependencies.
+- Download wazuh-qa repository , install requirements and package dependencies.
```
Comment (BraulioV) [inline docs/tests/integration/setting_up_test_environment.md:65]:
```suggestion
- Download wazuh-qa repository, install the requirements and the package dependencies.
```

File: docs/tests/integration/setting_up_test_environment.md [hunk 4]
```diff
@@ -114,15 +116,19 @@ echo 'monitord.rotate_log=0' >> "C:\Program Files (x86)\ossec-agent\local_intern
 
 - Restart **Wazuh** using the GUI
 
-- Download wazuh-qa repository and install package dependencies.
+- Download wazuh-qa repository , install requirements and package dependencies.
```
Comment (BraulioV) [inline docs/tests/integration/setting_up_test_environment.md:119]:
```suggestion
- Download wazuh-qa repository, install the requirements and the package dependencies.
```

File: docs/tests/integration/setting_up_test_environment.md [hunk 5]
```diff
@@ -114,15 +116,19 @@ echo 'monitord.rotate_log=0' >> "C:\Program Files (x86)\ossec-agent\local_intern
...
 
-```shell script
+```
 # Download wazuh-qa repository
 git clone https://github.com/wazuh/wazuh-qa.git
 
+# Install requirements
```
Comment (BraulioV) [inline docs/tests/integration/setting_up_test_environment.md:125]:
```suggestion
# Install the requirements
```

File: docs/tests/integration/setting_up_test_environment.md [hunk 6]
```diff
@@ -164,12 +169,15 @@ echo 'monitord.rotate_log=0' >> /Library/Ossec/etc/local_internal_options.conf
 /Library/Ossec/bin/ossec-control restart
 ```
 
-- Download wazuh-qa repository and install package dependencies.
+- Download wazuh-qa repository, install requirements and package dependencies.
```
Comment (BraulioV) [inline docs/tests/integration/setting_up_test_environment.md:172]:
```suggestion
- Download wazuh-qa repository, install the requirements and the package dependencies.
```

File: requirements.txt [hunk 7]
```diff
@@ -0,0 +1,20 @@
+pytest
+jq
+jsonschema
+psutil
+paramiko
+distro
+pandas
+freezegun
```
Comment (BraulioV) [inline requirements.txt:8]: You must set a version for these dependencies.

**Review by BraulioV — CHANGES_REQUESTED** [review]

**Review by BraulioV — APPROVED** [review]

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**